### DISCUSSION TITLE
Hello, how can I train or use model for russian language?

### DISCUSSION THREAD

**Discussion by javohirss** [mindee/doctr#1065] [Q&A] [answered]
I want to use this model. But it does not support Russian. How can I train model on custom dataset or something else? Is there any instruction?

**Comment by felixdittrich92** [comment, marked as answer]
Hi @javohirss :wave:,
1. you need to add russian to vocabs: https://github.com/mindee/doctr/blob/main/doctr/datasets/vocabs.py (feel free to open a PR to add it fixed :+1: )
2. train models on custom dataset instruction: 
detection: https://github.com/mindee/doctr/tree/main/references/detection
recognition: https://github.com/mindee/doctr/tree/main/references/recognition

**Reply by javohirss** [reply]
Do I need to train both recognition and detection? Because the detection suits me completely.
And about the data for recognition: every image must contain only one word?

**Reply by felixdittrich92** [reply]
If the detection works well in your case then no :)
yep each image only one word

**Reply by javohirss** [reply]
Excuse me, but what about the size of the images? Should they all be the same?

**Reply by felixdittrich92** [reply]
Can be different but a good range is height: 16-64 and width: 64-256 all images will be reshaped to 32x128 in the transformation

**Reply by javohirss** [reply]
thx a lot

**Reply by felixdittrich92** [reply]
you can take a look at the samples you pass in the model with `--show-samples`
